app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -143,11 +166,10 @@ object AppRepository {
         scheduleNetworkRepository.parseSchedule(scheduleXml, eTag,
                 onUpdateSessions = { sessions ->
                     val oldSessions = loadSessionsForAllDays(true)
-                    val newSessions = sessions.toSessionsAppModel2().sanitize()
-                    val hasChanged = ScheduleChanges.hasScheduleChanged(newSessions, oldSessions)
-                    if (hasChanged) {
+                    val newFlaggedSessions = sessions.toFlaggedSessions(oldSessions.toSessionsNetworkModel()) {
                         resetChangesSeenFlag()
                     }
+                    val newSessions = newFlaggedSessions.toSessionsAppModel2().sanitize()
```

Comment:
Why? Did you find an error?

Reply:
Ah! I see. This is how it was before. Thanks!

Reply:
For one, it's the way it was done before. If you change it, it should be on you to provide a reason :)

But if you go through the algorithm with some sample data that will be changed by the sanitization step you'll easily see why it's necessary:

1. Initially downloaded schedule contains a session with identical `title` and `subtitle` values. `sanitize()` will change the object so `subtitle` is the empty string. The cleaned up session is written to the database.
2. Download a new schedule. The session mentioned in step 1 hasn't changed, i.e. it's still `title == subtitle`.
3. `AppRepository` will read the old sessions from the database (with `subtitle` being the empty string). The compare function in `ScheduleChanges` will report a change because the `subtitle` of the new session object doesn't match that of the old session object.
4. `sanitize()` will rewrite the new session object to set the `subtitle` field to the empty string. Now the old and the new object are identical… but it has already been marked as changed.

Reply:
:heavy_check_mark: Changed in 8939692fa05a4edbea751479d2cfac82110af40e.